### PR TITLE
Added AutoAR

### PR DESCRIPTION
**Pull request by mloning**
#### Reference Issues/PRs
#245 

#### What does this implement/fix? Explain your changes.
* Adds AutoAR forecaster

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/master/.all-contributorsrc).
- [x] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/master/CODEOWNERS) to receive notifications about future changes to these files.
- [x] I've added unit tests and made sure they pass locally.

##### For new estimators
- [x] I've added the estimator to the online documentation.
- [ ] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

**Comment by Afzal-Ind** [comment]
Hi, could you tell more specifically what's to be done here next? Thanks.

**Comment by mloning** [comment]
Hi @Afzal-Ind - I'm not exactly sure what we have to adjust to make it work. You have to look into it and debug it I'm afraid. 

In the end, we want to be able to run something like this: 

```python
from sktime.forecasting.all import * 
y = load_airline()
f = AutoAR(maxlag=20)
f.fit(y)
fh = ForecastingHorizon([1, 2, 3])
y_pred = f.predict(fh)
```

File: sktime/forecasting/ar.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3 -u
+# -*- coding: utf-8 -*-
+
+__author__ = ["Markus Löning"]
+__all__ = ["AutoAR"]
+
+from statsmodels.tsa.ar_model import ar_select_order
+
+from sktime.forecasting.base._statsmodels import _StatsModelsAdapter
```
Comment (Afzal-Ind) [inline sktime/forecasting/ar.py:9]: Throwing this error while running: No module named 'sktime.forecasting.base._statsmodels'

Comment (mloning) [reply sktime/forecasting/ar.py:9]: The `_StatsModelsAdapter` class still exists, it moved into a different location. If you search for it, you'll find it! `sktime/forecasting/base/_adapters.py` perhaps?

**Comment by Afzal-Ind** [comment]
I will look into it. Thanks.

**Comment by afzal442** [comment]
> Hi @Afzal-Ind - I'm not exactly sure what we have to adjust to make it work. You have to look into it and debug it I'm afraid.
> 
> In the end, we want to be able to run something like this:
> 
> ```python
> from sktime.forecasting.all import * 
> y = load_airline()
> f = AutoAR(maxlag=20)
> f.fit(y)
> fh = ForecastingHorizon([1, 2, 3])
> y_pred = f.predict(fh)
> ```


`1961-01    454.177488  
1961-02    415.725546  
1961-03    447.075167`

@mloning This is the output that I am getting after debugging. Plz, review my PR #691. Thanks.

**Comment by mloning** [comment]
Closed in favour of #691

**Pull request closed without merging**